system/datatypes/mailinglist.php

```diff
@@ -1083,6 +1149,24 @@ protected function mailingSendTestAction(HyphaRequest $request) {
 		return 'reload';
 	}
 
+	protected function mailingProgressAction(HyphaRequest $request) {
+
+		$mailingId = $request->getArg(1);
+
+		// check if given id was correct
+		/** @var HyphaDomElement $mailing */
+		$mailing = $this->xml->getElementById($mailingId);
+		if (!$mailing instanceof HyphaDomElement) {
+			return '404';
+		}
+		$status = $mailing->getAttribute(self::FIELD_NAME_STATUS);
+		$progress = $mailing->getAttribute(self::FIELD_NAME_PROGRESS);
+		$total = $this->getNumberOfAddresses();
+		$result = [ self::FIELD_NAME_PROGRESS => $progress, self::FIELD_NAME_TOTAL => $total, self::FIELD_NAME_STATUS => $status];
+		echo json_encode($result);
```

Comment:
This should probably also set a Content-type header, to allow browsers and client frameworks to know that his is JSON. Not strictly required now, but probably helpful for debugging nonetheless.

Reply:
Done :)